### PR TITLE
[MESOS-10199] Fix Host field in client headers

### PR DESCRIPTION
**Pull request by zorro786**
This PR replaces empty host with actual IP/Port of the message recipient in Host field. IP+Port has been used as per RFC guidelines.
An earlier PR made host field empty https://reviews.apache.org/r/27865

### CODE REVIEW

**Comment by zorro786** [comment]
Ran `make check` in Linux. `HttpServeTest.Discard` test seems flaky, it fails with or without this change.
Also verified agent-master registration/communication behind nginx proxy.

**Comment by bmahler** [comment]
Thanks for fixing this, a couple of things, maybe @vinodkone or @asekretenko can have a look after these updates:

* Can you follow the existing logic for the http client?  See: https://github.com/apache/mesos/blob/1.11.0/3rdparty/libprocess/src/http.cpp#L1092-L1108
* Can you remove the extra .gitignore file? It's not related to this change and we like to keep our commits tidy with only a single logical change involved
* Can you add or update one of the existing tests in libprocess?

**Comment by zorro786** [comment]
> * Can you follow the existing logic for the http client?  See: https://github.com/apache/mesos/blob/1.11.0/3rdparty/libprocess/src/http.cpp#L1092-L1108

I find the existing logic redundant in this case, as `message.to.address` is an Inet address not a URL, which already includes an IP and port. `message.to.address` essentially includes the following code block already:

```
 else if (request.url.ip.isSome()) {
    headers["Host"] = stringify(request.url.ip.get());
  }
// Add port for non-standard ports.
if (request.url.port.isSome() &&
      request.url.port != 80 &&
      request.url.port != 443) {
    headers["Host"] += ":" + stringify(request.url.port.get());
  }
```
Please let me know if you meant something else.

> * Can you remove the extra .gitignore file? It's not related to this change and we like to keep our commits tidy with only a single logical change involved

Done
> * Can you add or update one of the existing tests in libprocess?

Done

cc: @vinodkone @asekretenko

**Comment by zorro786** [comment]
Is there any CI job which verifies build for my sanity? I haven't been successful in running all tests due to the flaky HttpServeTest.Discard

File: 3rdparty/libprocess/src/tests/process_tests.cpp [hunk 1]
```diff
@@ -1463,12 +1465,17 @@ TEST_F(ProcessTest, Remote)
...
   const string data = MessageEncoder::encode(message);
 
   AWAIT_READY(socket.send(data));
 
-  AWAIT_READY(handler);
+  AWAIT_READY(body);
+  ASSERT_EQ("hello world", body.get());
+
+  AWAIT_READY(pid);
+  ASSERT_EQ(message.from, pid.get());
```
Comment (vinodkone) [inline 3rdparty/libprocess/src/tests/process_tests.cpp:1478]: Would this assert fail if HOST header is not set with a real ip port?

**Comment by vinodkone** [comment]
@zorro786  If you believe HTTPDiscard test is flaky you can run the test suite by filter out that test (e.g., make check GTEST_FILTER=-"HttpServeTest.Discard")

File: 3rdparty/libprocess/src/tests/process_tests.cpp (hunk 1 above)
Comment (zorro786) [reply 3rdparty/libprocess/src/tests/process_tests.cpp:1478]: No it won't, it is just the pid of the sender process. Regardless of what the header is set this should succeed.

I actually wanted to write a Unit Test to verify host header being set at the receiver, however it is not straightforward to me from code to get host headers from libprocess' process. Message handlers in `process/process.hpp` seem to only care about process id and message body. I added the above as an Integration Test to verify connectivity/data transfer with changed header.

**Comment by zorro786** [comment]
Thanks @vinodkone. I have run the tests on master branch, two tests fail:
`HttpServeTest.Discard` and `HttpServeTest.Pipelining`.

Same tests fail on this PR change too. Rest all pass.

**Comment by vinodkone** [comment]
LGTM. I merged it. Thanks for the fix @zorro786

**Pull request merged**